geopandas/geodataframe.py

```diff
@@ -1676,41 +1676,28 @@ def explode(self, column=None, ignore_index=False, index_parts=None, **kwargs):
 
         df_copy = self.copy()
 
-        level_str = f"level_{df_copy.index.nlevels}"
-
-        if level_str in df_copy.columns:  # GH1393
-            df_copy = df_copy.rename(columns={level_str: f"__{level_str}"})
-
-        if index_parts:
-            exploded_geom = df_copy.geometry.explode(index_parts=True)
-            exploded_index = exploded_geom.index
-            exploded_geom = exploded_geom.reset_index(level=-1, drop=True)
-        else:
-            exploded_geom = df_copy.geometry.explode(index_parts=True).reset_index(
-                level=-1, drop=True
-            )
-            exploded_index = exploded_geom.index
-
-        df = (
-            df_copy.drop(df_copy._geometry_column_name, axis=1)
-            .join(exploded_geom)
-            .set_geometry(self._geometry_column_name)
-            .__finalize__(self)
+        exploded_geom = df_copy.geometry.reset_index(drop=True).explode(
+            index_parts=True
         )
 
+        df = geopandas.GeoDataFrame(
+            df_copy.drop(self._geometry_column_name, axis=1).take(
+                exploded_geom.reset_index(level=-1, drop=True).index
```

Comment:
```suggestion
                exploded_geom.index.droplevel(-1)
```

This might be a little bit more efficient, as it doesn't copy the Series with geometries just to get the new index